vm/src/memory/unified.rs

```diff
@@ -80,6 +80,16 @@ use std::{
 use super::{
     FixedMemory, LoadOp, MemAccessSize, MemoryProcessor, StoreOp, VariableMemory, NA, RO, RW, WO,
 };
+use crate::memory::fixed::FixedMemoryAddrValBytesIter;
+
+// Helper to cast &FixedMemory<RW> to &FixedMemory<RO>
+fn as_ro(mem: &FixedMemory<RW>) -> &FixedMemory<RO> {
+    unsafe { &*(mem as *const FixedMemory<RW> as *const FixedMemory<RO>) }
```

Comment:
As cryptographic software we have a very strong aversion to using `unsafe` wherever possible.

Moreover, is it not possible to just do

```
self.frw_store[idx].clone().into<FixedMemory<NA>>()
```

which will implicitly use the `From` implementation in `vm/src/memory/fixed.rs`? The compiler should do a good job of making the cloning/conversion quite fast, and though slower than the recast approach is I think worth doing to avoid any `unsafe` issues.